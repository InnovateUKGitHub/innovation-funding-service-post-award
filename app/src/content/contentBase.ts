import i18next from "i18next";
import { ProjectDto } from "@framework/dtos";

export interface ContentResult {
  readonly key: string;
  readonly content: string;
  readonly markdown: boolean;
}

export interface MarkdownOption { markdown?: boolean; }
export interface DataOption { [key: string]: any; }

export abstract class ContentBase {
  constructor(private readonly parent: ContentBase | null, private readonly name: string | null, protected project?: ProjectDto | null | undefined) {
  }

  protected getNameParts(): string[] {
    const result = this.parent ? this.parent.getNameParts() : [];
    if (this.name) {
      result.push(this.name);
    }
    return result;
  }

  protected getContent(key: string, options?: MarkdownOption & DataOption): ContentResult {
    const { markdown } = options || { markdown: false };
    const parts = this.getNameParts();
    key.split(".").forEach(x => parts.push(x));
    const initialKey = parts.join(".");

    while (parts.length) {
      const fullKey = parts.join(".");
      const contentResult = this.getCustomContent(fullKey, options);
      if (!!contentResult) return contentResult;
      parts.shift();
    }
    return {
      key: initialKey,
      content: initialKey,
      markdown: markdown || false
    };
  }

  private getCustomContent(fullKey: string, options?: MarkdownOption & DataOption): ContentResult | undefined {
    const { markdown, ...i18tOptions } = options || { markdown: false };
    if (this.project) {
      const fullKeyWithCompetitionType = `${this.project.competitionType}.${fullKey}`;
      if (i18next.exists(fullKeyWithCompetitionType)) {
        return {
          key: fullKeyWithCompetitionType,
          content: i18next.t(fullKeyWithCompetitionType, i18tOptions),
          markdown: markdown || false
        };
      }
    }

    if (i18next.exists(fullKey)) {
      return {
        key: fullKey,
        content: i18next.t(fullKey, i18tOptions),
        markdown: markdown || false
      };
    }
  }
}
