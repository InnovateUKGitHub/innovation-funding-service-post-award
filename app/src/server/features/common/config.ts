import bytes from "bytes";
import { parseLogLevel } from "@framework/types/logLevel";
import { parseAccEnvironment } from "@framework/types/accEnvironment";
import { IConfig } from "@framework/types/IConfig";
import { AccEnvironment, LogLevel } from "@framework/constants/enums";
import {
  getBooleanEnv as boolEnv,
  getCertificateEnv as certEnv,
  getCommaSepStringArray as csvEnv,
  getEnv as env,
  getFloatEnv as floatEnv,
  getIntegerEnv as intEnv,
  getPipeSepStringArray as psvEnv,
  getStringEnv as strEnv,
} from "@framework/util/envHelpers";

const configuration: Readonly<IConfig> = {
  build: strEnv("BUILD", String(Date.now())),
  salesforceQueryLimit: intEnv("SALESFORCE_QUERY_LIMIT", 195),
  disableCsp: boolEnv("DISABLE_CSP", false),
  cookie: {
    secret: strEnv("COOKIE_KEY"),
    secure: boolEnv("COOKIE_SECURE", true),
  },
  timeouts: {
    costCategories: floatEnv("COST_CAT_TIMEOUT_MINUTES", 720),
    projectRoles: floatEnv("PROJ_ROLES_TIMEOUT_MINUTES", 720),
    recordTypes: floatEnv("RECORD_TYPES_TIMEOUT_MINUTES", 720),
    optionsLookup: floatEnv("OPTIONS_LOOKUP_TIMEOUT_MINUTES", 720),
    token: floatEnv("TOKEN_TIMEOUT_MINUTES", 10),
    cookie: floatEnv("COOKIE_TIMEOUT_MINUTES", 30),
    contentRefreshSeconds: floatEnv("CONTENT_REFRESH_TIMEOUT_SECONDS", 0),
  },
  certificates: {
    salesforce: certEnv("SALESFORCE_PRIVATE_KEY"),
    saml: {
      idp: {
        public: certEnv("SHIBBOLETH_IDP_PUBLIC_KEY", ""),
      },
      spSigning: {
        public: certEnv("SHIBBOLETH_SP_SIGNING_PUBLIC_KEY", ""),
        private: certEnv("SHIBBOLETH_SP_SIGNING_PRIVATE_KEY", ""),
      },
      spDecryption: {
        public: certEnv("SHIBBOLETH_SP_DECRYPTION_PUBLIC_KEY", ""),
        private: certEnv("SHIBBOLETH_SP_DECRYPTION_PRIVATE_KEY", ""),
      },
    },
  },
  basicAuth: {
    credentials: psvEnv("BASIC_AUTH", []),
  },
  features: {
    changePeriodLengthWorkflow: boolEnv("FEATURE_CHANGE_PERIOD_LENGTH", false),
    customContent: boolEnv("FEATURE_CUSTOM_CONTENT", false),
    searchDocsMinThreshold: intEnv("SEARCH_DOCS_MIN_THRESHOLD", 5),
    futureTimeExtensionInYears: intEnv("FUTURE_TIME_EXTENSION_IN_YEARS", 5),
    approveNewSubcontractor: boolEnv("FEATURE_APPROVE_NEW_SUBCONTRACTOR", false),
  },
  logLevel: env("LOG_LEVEL", LogLevel.ERROR, parseLogLevel),
  options: {
    bankCheckAddressScorePass: intEnv("BANK_CHECK_ADDRESS_SCORE_PASS", 6),
    bankCheckValidationRetries: intEnv("BANK_CHECK_VALIDATION_RETRIES", 3),
    bankCheckCompanyNameScorePass: intEnv("BANK_CHECK_COMPANY_NAME_SCORE_PASS", 6),
    permittedTypes: {
      pdfTypes: csvEnv("PERMITTED_DOCUMENT_FILE_TYPES", ["pdf", "xps"]),
      textTypes: csvEnv("PERMITTED_WORD_PROCESSING_FILE_TYPES", ["doc", "docx", "rtf", "txt", "odt"]),
      presentationTypes: csvEnv("PERMITTED_PRESENTATION_FILE_TYPES", ["ppt", "pptx", "odp"]),
      spreadsheetTypes: csvEnv("PERMITTED_SPREADSHEET_FILE_TYPES", ["csv", "xls", "xlsx", "ods"]),
      imageTypes: csvEnv("PERMITTED_IMAGE_FILE_TYPES", ["jpg", "jpeg", "png", "odg"]),
    },
    maxUploadFileCount: intEnv("MAX_UPLOAD_FILE_COUNT", 10),
    maxFileSize: intEnv("MAX_FILE_SIZE_IN_BYTES", bytes("32MB")),
    maxTotalFileSize: intEnv("MAX_TOTAL_FILE_SIZE_IN_BYTES", bytes("32MB")),
    maxFileBasenameLength: intEnv("MAX_FILE_BASENAME_LENGTH", 80),
    standardOverheadRate: floatEnv("STANDARD_OVERHEAD_RATE", 20),
    numberOfProjectsToSearch: intEnv("FEATURE_SEARCH_NUMBER_PROJECTS", 3),
    maxClaimLineItems: intEnv("FEATURE_MAX_CLAIM_LINE_ITEMS", 120),
    nonJsMaxClaimLineItems: intEnv("NON_JS_FEATURE_MAX_CLAIM_LINE_ITEMS", 120),
  },
  salesforceServiceUser: {
    clientId: strEnv("SALESFORCE_CLIENT_ID"),
    connectionUrl: strEnv("SALESFORCE_CONNECTION_URL"),
    serviceUsername: strEnv("SALESFORCE_USERNAME"),
  },
  bankDetailsValidationUser: {
    clientId: strEnv("SALESFORCE_CLIENT_ID"),
    connectionUrl: strEnv("SALESFORCE_CONNECTION_URL"),
    serviceUsername: strEnv("SALESFORCE_BANK_DETAILS_VALIDATION_USERNAME"),
  },
  companiesHouse: {
    endpoint: strEnv("COMPANIES_HOUSE_ENDPOINT", ""),
    accessToken: strEnv("COMPANIES_HOUSE_ACCESS_TOKEN", ""),
  },
  sil: {
    url: strEnv("SIL_HYDRA_URL", ""),
  },
  sso: {
    enabled: boolEnv("USE_SSO", false),
    providerUrl: strEnv("SSO_PROVIDER_URL", ""),
    signoutUrl: strEnv("SSO_SIGNOUT_URL", ""),
  },
  urls: {
    ifsRoot: strEnv("IFS_ROOT", "https://apply-for-innovation-funding.service.gov.uk"),
    ifsApplicationUrl: strEnv(
      "IFS_APPLICATION_URL",
      "https://application-for-innovation-funding.service.gov.uk/management/competition/<<Acc_CompetitionId__c>>/application/<<Acc_IFSApplicationId__c>>",
    ),
    ifsGrantLetterUrl: strEnv(
      "IFS_GRANT_LETTER_URL",
      "https://application-for-innovation-funding.service.gov.uk/management/competition/<<Acc_CompetitionId__c>>/project/<<Acc_IFSApplicationId__c>>",
    ),
  },
  googleTagManagerCode: strEnv("GOOGLE_TAG_MANAGER_CODE", ""),
  s3Account: {
    accessKeyId: strEnv("AWS_S3_ACCESS_KEY_ID", ""),
    secretAccessKey: strEnv("AWS_S3_ACCESS_SECRET", ""),
    contentBucket: strEnv("AWS_S3_CONTENT_BUCKET", ""),
    customContentPath: strEnv("AWS_S3_CONTENT_CUSTOM_PATH", ""),
  },
  accEnvironment: env("ACC_ENVIRONMENT", AccEnvironment.UNKNOWN, parseAccEnvironment),
  newRelic: {
    enabled: boolEnv("NEW_RELIC_ENABLED", false),
    eventsUrl: strEnv("NEW_RELIC_EVENTS_URL", ""),
    apiKey: strEnv("NEW_RELIC_API_KEY", ""),
    appName: strEnv("NEW_RELIC_APP_NAME", ""),
  },
  webserver: {
    port: strEnv("PORT", "8080"),
    url: strEnv("SERVER_URL"),
  },
  developer: {
    writeGraphQL: boolEnv("DEVELOPER_WRITE_GRAPHQL", false),
  },
};

export { configuration };
