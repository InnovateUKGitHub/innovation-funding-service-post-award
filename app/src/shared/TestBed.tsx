import { createMemoryHistory } from "history";
import _merge from "lodash.merge";
import { Router } from "react-router-dom";
import { Copy } from "@copy/Copy";
import { mountedContext } from "@ui/components/atomicDesign/atoms/providers/Mounted/Mounted";
import { PageTitleProvider } from "@ui/features/page-title";
import { ContentProvider } from "@ui/redux/contentProvider";
import { IStores, StoresProvider } from "@ui/redux/storesProvider";
import { ClientConfigProvider } from "@ui/components/providers/ClientConfigProvider";
import { IClientConfig } from "src/types/IClientConfig";
import { AccEnvironment } from "@framework/constants/enums";

export type TestBedStore = Partial<IStores>;

export interface ITestBedProps {
  children: React.ReactElement;
  competitionType?: string;
  stores?: TestBedStore;
  isServer?: boolean;
  pageTitle?: string;
  /**
   * `shouldOmitRouterProvider` will prevent a router provider being added to the test bed.
   * Because there can only be one router provider in the tree, it may happen that in
   * some cases there will need to be a separate router provider, in which case pass this flag
   * in to enable building.
   */
  shouldOmitRouterProvider?: boolean;
  extendClientConfig?: (config: RecursiveMutable<IClientConfig>) => void;
}

/**
 * TestBed is a component configured with all stores needed for tests to run
 *
 * Note: When testing a component that consumes <Content />, it expects a provider to be present.
 */
export function TestBed({
  isServer = false,
  stores,
  competitionType,
  children,
  pageTitle = "stub-displayTitle",
  shouldOmitRouterProvider,
  extendClientConfig,
}: ITestBedProps) {
  const stubStores = {
    users: {
      getCurrentUser: () => ({ csrf: "stub-csrf" }),
    },
    navigation: {
      getPageTitle: () => ({
        displayTitle: "stub-displayTitle",
      }),
    },
  };

  const storesValue = _merge(stubStores, stores) as Required<TestBedStore>;

  const clientConfig = {
    features: {
      changePeriodLengthWorkflow: false,
      customContent: false,
      searchDocsMinThreshold: 9999,
      futureTimeExtensionInYears: 5,
    },
    options: {
      maxFileSize: 33554432,
      maxUploadFileCount: 10,
      maxFileBasenameLength: 80,
      permittedFileTypes: [
        "pdf",
        "xps",
        "doc",
        "docx",
        "rtf",
        "txt",
        "csv",
        "odt",
        "ppt",
        "pptx",
        "odp",
        "xls",
        "xlsx",
        "ods",
        "jpg",
        "jpeg",
        "png",
        "odg",
      ],
      permittedTypes: {
        pdfTypes: ["pdf", "xps"],
        textTypes: ["doc", "docx", "rtf", "txt", "csv", "odt"],
        presentationTypes: ["ppt", "pptx", "odp"],
        spreadsheetTypes: ["xls", "xlsx", "ods"],
        imageTypes: ["jpg", "jpeg", "png", "odg"],
      },
      bankCheckValidationRetries: 3,
      bankCheckAddressScorePass: 6,
      bankCheckCompanyNameScorePass: 6,
      standardOverheadRate: 20,
      numberOfProjectsToSearch: 3,
      maxClaimLineItems: 120,
      nonJsMaxClaimLineItems: 10,
    },
    ifsRoot: "https://ifs-accdev.apps.ocp4.org.innovateuk.ukri.org",
    ssoEnabled: false,
    logLevel: "VERBOSE",
    accEnvironment: AccEnvironment.BASE,
  };

  // Modify the clientConfig if the TextBed wants to.
  extendClientConfig?.(clientConfig);

  // Note: We need a way of upfront toggling this can use 'MountedProvider'
  const testBedMountState = { isServer, isClient: !isServer };

  const history = createMemoryHistory();

  const Providers = (
    <mountedContext.Provider value={testBedMountState}>
      <ClientConfigProvider config={clientConfig}>
        <PageTitleProvider title={pageTitle}>
          <StoresProvider value={storesValue}>
            <ContentProvider value={new Copy({ competitionType })}>{children}</ContentProvider>
          </StoresProvider>
        </PageTitleProvider>
      </ClientConfigProvider>
    </mountedContext.Provider>
  );

  return shouldOmitRouterProvider ? (
    Providers
  ) : (
    <Router location={history.location} navigator={history}>
      {Providers}
    </Router>
  );
}

export type HookTestBedProps = Omit<ITestBedProps, "children">;

export const hookTestBed = (props: HookTestBedProps) => ({
  wrapper: (wrapperProps: ITestBedProps) => <TestBed {...props} {...wrapperProps} />,
});

export default TestBed;
