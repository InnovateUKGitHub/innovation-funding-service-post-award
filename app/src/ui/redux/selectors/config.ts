import { RootState } from "@ui/redux";

export const getMaxFileSize = (state: RootState) => state.config.maxFileSize;
