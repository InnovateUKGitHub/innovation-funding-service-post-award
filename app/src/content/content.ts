import { ContentBase, ContentResult } from "./contentBase";
import { HomePageContent } from "./pages/homePageContent";
import { ProjectDashboardContent } from "./pages/projectDashboardContent";
import { ProjectOverviewContent } from "./pages/projectOverviewContent";

export type ContentSelector = (content: Content) => ContentResult;

export class Content extends ContentBase {
  constructor() {
    super(null, null);
  }

  readonly home = new HomePageContent(this);
  public projectsDashboard = new ProjectDashboardContent(this);
  public projectOverview = new ProjectOverviewContent(this);

}
