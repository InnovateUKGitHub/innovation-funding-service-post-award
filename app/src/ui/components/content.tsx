import { useContent } from "@ui/hooks";
import { ContentSelector } from "@content/content";
import { useStores } from "@ui/redux";
import { Modal } from "@ui/components/modal";
import { Section } from "@ui/components/layout";
import { SummaryList, SummaryListItem } from "@ui/components/summaryList";
import { ModalLink } from "@ui/components/links";
import { Markdown } from "./renderers/markdown";

interface IContentProps {
  value: ContentSelector;
  styles?: React.CSSProperties;
}

export const Content = ({ value, styles }: IContentProps) => {
  const stores = useStores();
  const { getResultByQuery } = useContent();

  const config = stores.config.getConfig().features;
  const { key, content, markdown } = getResultByQuery(value);

  const displayValue: string | React.ReactElement = markdown ? (
    <Markdown style={styles && { color: styles.color }} value={content} />
  ) : (
    content
  );

  const renderContentHint = () => {
    const modalId = `content-${key}`;

    return (
      <>
        &nbsp;
        <i
          aria-label="Open content modal dialogue"
          role="button"
          className="content-prompt__icon"
          tabIndex={0}
          onKeyDown={e => {
            e.preventDefault();
            e.stopPropagation();
            window.location.hash = `#${modalId}`;
          }}
        />
        <Modal id={modalId}>
          <Section title="Content hint">
            <SummaryList qa="content_info">
              <SummaryListItem label="Current value" content={displayValue} qa="current_value" />

              <SummaryListItem label="Content key" content={key} qa="current_value" />
            </SummaryList>

            <ModalLink styling="PrimaryButton" modalId={modalId} open={false}>
              Close
            </ModalLink>
          </Section>
        </Modal>
      </>
    );
  };

  return (
    <>
      {displayValue}
      {config.contentHint && renderContentHint()}
    </>
  );
};
