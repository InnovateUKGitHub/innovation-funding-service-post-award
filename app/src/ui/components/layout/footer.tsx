import cx from "classnames";

import { IExternalAsLink, ISupplementLinksOptions } from "@ui/containers/app/footer.config";

import { H3, UL, GovWidthContainer, CrownCopyrightLogo } from "@ui/components";
import { ExternalLink, ExternalLinkProps } from "@ui/components/renderers";

export interface FooterProps {
  footerContent: ISupplementLinksOptions;
  supportingLinks: IExternalAsLink[];
}

export function Footer({ supportingLinks, footerContent }: FooterProps) {
  const { externalContent, externalLinks } = footerContent;

  return (
    <footer className="govuk-footer" role="contentinfo">
      <GovWidthContainer>
        <div className="govuk-footer__navigation">
          <div className="govuk-footer__section govuk-grid-column-two-thirds">
            <H3 as="h2" className="govuk-footer__heading">
              {externalContent.title.content}
            </H3>

            <ul className="govuk-footer__list govuk-footer__list--columns-2">
              {supportingLinks.map(({ content, ...linkProps }) => (
                <li key={linkProps.id} className="govuk-footer__list-item">
                  <FooterLink {...linkProps}>{content}</FooterLink>
                </li>
              ))}
            </ul>
          </div>
        </div>

        <hr className="govuk-footer__section-break" />

        <div className="govuk-footer__meta">
          <div className="govuk-footer__meta-item govuk-footer__meta-item--grow">
            <UL className="govuk-footer__inline-list">
              <li className="govuk-footer__inline-list-item" data-qa={externalContent.usesCookie.id}>
                {externalContent.usesCookie.content}
              </li>

              <li className="govuk-footer__inline-list-item">
                <FooterLink id={externalLinks.moreAboutCookies.id} href={externalLinks.moreAboutCookies.href}>
                  {externalLinks.moreAboutCookies.content}
                </FooterLink>
                .
              </li>
            </UL>

            <CrownCopyrightLogo
              className="govuk-footer__licence-logo"
              height="17"
              width="41"
              role="presentation"
              focusable="false"
            />

            <span className="govuk-footer__licence-description" data-qa="licence">
              <span data-qa={externalContent.govLicenseLinkPart1.id}>
                {externalContent.govLicenseLinkPart1.content}
              </span>

              <FooterLink
                rel="license"
                id={externalLinks.govLicenseLinkPart2.id}
                href={externalLinks.govLicenseLinkPart2.href}
              >
                {externalLinks.govLicenseLinkPart2.content}
              </FooterLink>

              <span data-qa={externalContent.govLicenseLinkPart3.id}>
                {externalContent.govLicenseLinkPart3.content}
              </span>
            </span>
          </div>
          <div className="govuk-footer__meta-item">
            <FooterLink
              className="govuk-footer__copyright-logo"
              id={externalLinks.crownCopyrightLink.id}
              href={externalLinks.crownCopyrightLink.href}
            >
              {externalLinks.crownCopyrightLink.content}
            </FooterLink>
          </div>
        </div>
      </GovWidthContainer>
    </footer>
  );
}

function FooterLink({ className, ...props }: ExternalLinkProps) {
  return <ExternalLink data-qa={props.id} className={cx("govuk-footer__link", className)} {...props} />;
}
