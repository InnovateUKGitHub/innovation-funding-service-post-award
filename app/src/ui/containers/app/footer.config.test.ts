import { renderHook } from "@testing-library/react";
import { hookTestBed } from "@shared/TestBed";
import { footerLinks, useFooterExternalContent } from "@ui/containers/app/footer.config";
import { Content } from "@content/content";

describe("Footer config", () => {
  describe("footerLinks", () => {
    test("as default", () => {
      expect(footerLinks).toMatchSnapshot();
    });
  });

  describe("useFooterExternalContent", () => {
    const setup = () => {
      const stubContent = {
        footer: {
          supportLinks: { content: "stub-content-supportLinks" },
          explainCookies: { content: "stub-content-explainCookies" },
          cookieFindOutMore: { content: "stub-content-cookieFindOutMore" },
          govLicenseLinkPart1: { content: "stub-content-govLicenseLinkPart1" },
          govLicenseLinkPart2: { content: "stub-content-govLicenseLinkPart2" },
          govLicenseLinkPart3: { content: "stub-content-govLicenseLinkPart3" },
          crownCopyright: { content: "stub-content-crownCopyright" },
        },
      } as Partial<Content>;

      const { result } = renderHook(useFooterExternalContent, hookTestBed({ content: stubContent }));

      return result.current;
    };

    test("should return with content", () => {
      const footerContent = setup();

      expect(footerContent).toMatchSnapshot();
    });
  });
});
