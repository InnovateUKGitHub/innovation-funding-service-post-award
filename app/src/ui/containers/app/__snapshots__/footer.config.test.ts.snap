// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer config footerLinks as default 1`] = `
Array [
  Object {
    "content": "Innovate UK",
    "href": "https://www.gov.uk/government/organisations/innovate-uk",
    "id": "innovate-uk",
  },
  Object {
    "content": "Innovation funding advice",
    "href": "https://www.gov.uk/guidance/innovation-apply-for-a-funding-award",
    "id": "innovation-funding-advice",
  },
  Object {
    "content": "Connect to innovation experts",
    "href": "https://www.gov.uk/guidance/innovation-get-support-and-advice",
    "id": "connect-to-innovation-experts",
  },
  Object {
    "content": "Events",
    "href": "https://connect.innovateuk.org/events",
    "id": "events",
  },
  Object {
    "content": "Innovate UK blog",
    "href": "https://innovateuk.blog.gov.uk/",
    "id": "innovate-uk-blog",
  },
  Object {
    "content": "GOV.UK accessibility",
    "href": "https://www.gov.uk/help/accessibility",
    "id": "gov.uk-accessibility",
  },
  Object {
    "content": "Terms and conditions",
    "href": "https://apply-for-innovation-funding.service.gov.uk/info/terms-and-conditions",
    "id": "terms-and-conditions",
  },
  Object {
    "content": "Contact us",
    "href": "https://apply-for-innovation-funding.service.gov.uk/info/contact",
    "id": "contact-us",
  },
  Object {
    "content": "Sign up for competition updates",
    "href": "http://info.innovateuk.org/emailpref",
    "id": "sign-up-for-competition-updates",
  },
  Object {
    "content": "Latest funding opportunities",
    "href": "https://apply-for-innovation-funding.service.gov.uk/competition/search",
    "id": "latest-funding-opportunities",
  },
]
`;

exports[`Footer config useFooterExternalContent should return with content 1`] = `
Object {
  "externalContent": Object {
    "govLicenseLinkPart1": Object {
      "content": "stub-content-govLicenseLinkPart1",
      "id": "govLicenseLinkPart1",
    },
    "govLicenseLinkPart3": Object {
      "content": "stub-content-govLicenseLinkPart3",
      "id": "govLicenseLinkPart3",
    },
    "title": Object {
      "content": "stub-content-supportLinks",
      "id": "footer-title",
    },
    "usesCookie": Object {
      "content": "stub-content-explainCookies",
      "id": "uses-cookies",
    },
  },
  "externalLinks": Object {
    "crownCopyrightLink": Object {
      "content": "stub-content-crownCopyright",
      "href": "https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/",
      "id": "crown-copyright-link",
    },
    "govLicenseLinkPart2": Object {
      "content": "stub-content-govLicenseLinkPart2",
      "href": "https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/",
      "id": "license-link",
    },
    "moreAboutCookies": Object {
      "content": "stub-content-cookieFindOutMore",
      "href": "https://apply-for-innovation-funding.service.gov.uk/info/cookies",
      "id": "more-about-cookies-text",
    },
  },
}
`;